lms/services/group_info.py

```diff
@@ -46,7 +46,7 @@ def upsert_group_info(self, grouping: Grouping, params: dict):
         # AI should be many:many, and we reflect that wrongness here.
         group_info.application_instance = grouping.application_instance
 
-        group_info.type = self._GROUPING_TYPES[grouping.type]
+        group_info.type = self._GROUPING_TYPES[grouping.type]  # type: ignore
```

Comment:
We don't really maintain group_info that much. THe fix here is to refactor it completely or remove it.